### ISSUE TITLE
Replace Ironic stub images with Ironic built from RDO

### ISSUE THREAD

**Issue by vrutkovs** [openshift/okd#197]
Currently ironic and baremetal-* images are not being synced into 4.x imagestream in `origin` - https://github.com/openshift/release/blob/master/core-services/release-controller/_releases/release-ocp-4.4-ci.json#L25-L32. These might have been built using OSP repos, so we can't use these in OKD.

Operators should use RDO repos to build openstack-related and promote these images to `origin` namespace (perhaps from metal3 builds?)

The following images need to be reviewed:
* [ ] `baremetal-machine-controllers`
      https://github.com/openshift/cluster-api-provider-baremetal/blob/master/Dockerfile. @dhellmann @hardys
* [ ] `baremetal-operator`
      https://github.com/openshift/baremetal-operator/blob/master/Dockerfile @honza @stbenjam 
* [ ] `ironic`
      https://github.com/openshift/ironic-image/blob/master/Dockerfile.ocp @derekhiggins @dtantsur
* [ ] `ironic-inspector`
      https://github.com/openshift/ironic-inspector-image/blob/master/Dockerfile.ocp @elfosardo @stbenjam 
* [ ] `ironic-ipa-downloader`
      https://github.com/openshift/ironic-ipa-downloader/blob/master/Dockerfile.ocp @derekhiggins @elfosardo 
* [ ] `ironic-machine-os-downloader`
      https://github.com/openshift/ironic-rhcos-downloader/blob/master/Dockerfile.ocp @hardys @elfosardo 
* [ ] `ironic-static-ip-manager`
      https://github.com/openshift/ironic-static-ip-manager/blob/master/Dockerfile.ocp @elfosardo @russellb 
* [ ] any other image built with OSP repos and synced to origin

**Comment by dtantsur** [comment]
Can we just use the metal3 upstream images? They rely on RDO already.

**Comment by vrutkovs** [comment]
That would be perfect. Does our CI build those too?

**Comment by dhellmann** [comment]
> That would be perfect. Does our CI build those too?
> 
> EDIT: I don't think it does, so ideally we'd use the same code as in OCP but built using a different openstack repo

The images in the https://quay.io/organization/metal3-io org are built by quay.io when changes merge into the GitHub repos under https://github.com/metal3-io/. 

The quay version of the cluster-api-provider-baremetal image hasn't been rebuilt since March 23 because that component is now only used by OpenShift, so we use an internal build for that image.

**Comment by vrutkovs** [comment]
>The images in the quay.io/organization/metal3-io org are built by quay.io when changes merge into the GitHub repos under @metal3-io.

That's not suitable for OKD. Our release model is the same as OCP - image gets built on CI, but being promoted to `origin` namespace.

Could metal3.io folks help me with the following:
* review dockerfiles from the description. If these are using RHOSP Ironic we should create Dockerfile.okd which uses Ironic RDO (same for any other non-community component there).
* once all the repos have OKD-compatible dockerfile I can create PRs to release to build OKD images and promote them to `origin` namespace - I'm going to need `/lgtm` there

**Comment by hroyrh** [comment]
I found upstream counterparts inside metal3-io quay repository for most of these images, except :
-  baremetal-machine-controllers : the upstream repo it was forked from has been archived and a new upstream repo was created called [cluster-api-provider-metal3](https://github.com/metal3-io/cluster-api-provider-metal3)
- ironic-static-ip-manager has an upstream image but no upstream repo and hasn't been updated in a long time
- ironic-machine-os-downloader

Is someone working on building RDO based images, maybe we can reuse the existing ones from metal3 and rebuild the rest.

**Comment by vrutkovs** [comment]
Seems `ironic-static-ip-manager` and `ironic-machine-os-downloader` are not using RHOSP content, so we can start syncing them (although I'd like a confirmation from Metal3 folks anyway).

**Comment by dhellmann** [comment]
> * baremetal-machine-controllers : the upstream repo it was forked from has been archived and a new upstream repo was created called [cluster-api-provider-metal3](https://github.com/metal3-io/cluster-api-provider-metal3)

The upstream version is not compatible with OpenShift because it uses a newer version of the Machine API than OpenShift does. I would expect the same to apply to OKD. The openshift/cluster-api-provider-baremetal repository holds the version of that controller used in OpenShift.

**Comment by hardys** [comment]
> Seems `ironic-static-ip-manager` and `ironic-machine-os-downloader` are not using RHOSP content, so we can start syncing them (although I'd like a confirmation from Metal3 folks anyway).

That's correct, both of those images are just wrappers for scripts related to how we deploy ironic/metal3 in OpenShift, but with no dependency on RHOSP content.

Then we can work on adding the necesary Dockerfile.okd for remaining repos, which should be very similar to the upstream metal3-io Dockerfiles which consume RDO content.

**Comment by hardys** [comment]
> I found upstream counterparts inside metal3-io quay repository for most of these images, except :
>     * ironic-static-ip-manager has an upstream image but no upstream repo and hasn't been updated in a long time

https://github.com/metal3-io/static-ip-manager-image is the upstream repo but as you say it hasn't been updated in some time, because the metal3 community have moved to a different model for IP management.

Probably for OKD we should just rebuild https://github.com/openshift/ironic-static-ip-manager and we can discuss upstream if it makes sense to retire the repo mentioned above.
 
>     * ironic-machine-os-downloader

This is a downstream-only image, since it's related to preparing the RHCOS/FCOS image for deployment in the OpenShift deployment of metal3

Upstream an image URL is just provided directly (we can't do this downstream because the OS images are gz/xz compressed, which Ironic doesn't yet handle ref https://storyboard.openstack.org/#!/story/2006936)

**Comment by LorbusChris** [comment]
https://github.com/openshift/ironic-image/pull/123
https://github.com/openshift/ironic-ipa-downloader/pull/56

**Comment by vrutkovs** [comment]
https://github.com/openshift/release/pull/16377 to sync `baremetal-machine-controllers` and `baremetal-operator`

**Comment by vrutkovs** [comment]
https://github.com/openshift/release/pull/16379 to run OKD-specific build on Ironic image
